pages/forums/list.js

```diff
@@ -0,0 +1,208 @@
+import Page from '../../components/Page'
+import Link from '../../components/Link'
+
+
+
+
+
+// Component constants
+const pageTitle = 'Public Forums'
+
+
+
+
+
+class ForumList extends Component {
+  /***************************************************************************\
+    Private Methods
+  \***************************************************************************/
+
+  async _getThreads(page, oldPage) {
+    const { getForumThreads } = this.props
+
+    if (page !== oldPage) {
+      this.setState({ loaded: false })
+
+      const { payload, status } = await getForumThreads(page)
+
+      if (status === 'success') {
+        this.setState({
+          ...payload.meta,
+          loaded: true,
+          threads: payload.data,
+          totalPages: Math.ceil(payload.meta.count / payload.meta.limit),
+        })
+        return
+      }
+    }
+
+    this.setState({
+      loaded: true,
+      threads: null,
+    })
+  }
+
+  /***************************************************************************\
+    Public Methods
+  \***************************************************************************/
+
+  componentDidMount () {
+    const {
+      page,
+    } = this.props
+
+    this._getThreads(page, null)
+  }
+
+  componentWillReceiveProps(newProps) {
+    this._getThreads(newProps.page, this.props.page)
+  }
+
+  constructor(props) {
+    super(props)
+
+    this._bindMethods([
+      '_getThreads',
+    ])
+
+    this.state = {
+      count: 0,
+      limit: 0,
+      loaded: false,
+      offset: 0,
+      threads: [],
+      total: 0,
+      totalPages: 0,
+    }
+  }
+
+  render() {
+    if (!this.state.loaded) {
+      return (
+        <React.Fragment>
+          <header>
+            <h1>Public Forums</h1>
+          </header>
+
+          <span>Loading...</span>
+
+        </React.Fragment>
+      )
+    }
+
+    const {
+      count,
+      limit,
+      offset,
+      threads,
+      total,
+      totalPages,
+    } = this.state
+
+    const { page } = this.props
+
+    return (
+      <React.Fragment>
+        <header>
+          <h1>Public Forums</h1>
+
+          <menu type="toolbar">
+            <Link
+              action="create"
+              category="Forums"
+              label="Thread"
+              route="forum thread create">
+              <a
+                className="button success" >
+                New Thread
+              </a>
+            </Link>
+          </menu>
+        </header>
+
+
+        {!threads.length && (
+          <span>There is nothing here. (yet!!)</span>
+        )}
+
+        {!!threads.length && (
+          <div className="thread-list">
+            <span className="list-stats">Displaying threads {offset + 1} - {Math.min(count + offset, limit + offset)} of {total} threads</span>
+            {threads.map(thread => (
+              <ForumThreadCard thread={thread} key={thread.id} />
+            ))}
+          </div>
+        )}
+
+        <menu type="toolbar">
```

Comment:
We have a `<Pagination />` component for this!

Reply:
oh sweet! missed that